endpoints.php

```diff
@@ -1491,4 +1507,65 @@ function getQuestionHasAnswers($request)
         return new WP_Error($e->getCode(), $e->getMessage(), $e->getStatus());
     }
     
+}
+
+function upload_videos_to_kaltura()
+{
+    if (LOAD_KALTURA) {
+        $upload_folder = wp_upload_dir()['path'];
+    
+        $current_date = date('Y/m/d');
+
+        $tapestries = get_posts(['post_type' => 'tapestry',]);
+        $video_nodes = array();
+
+        $kalturaApi = new KalturaApi();
+
+        foreach ($tapestries as $value) {
+            $tapestry = new Tapestry($value->ID);
+
+            foreach ($tapestry->getNodeIds() as $node_id) {
+                $node = new TapestryNode($value->ID, $node_id);
+                $nodeMeta = $node->getMeta();
+
+                if ($nodeMeta->mediaType == "video") {
+                    $media_url = $node->getTypeData()->mediaURL;
+                    if (strpos($media_url, "/wp-content/uploads/")) {
```

Comment:
It's generally frowned upon to hardcode the upload directory. Use wp_upload_dir function instead.